core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/multitenancy/utils/TenantAxisUtils.java

```diff
@@ -455,8 +455,15 @@ public static void cleanupTenants(long tenantIdleTimeMillis) {
                                 PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
                                 carbonContext.setTenantDomain(tenantDomain, true);
 
-                                // Terminating idle tenant configuration contexts.
-                                terminateTenantConfigContext(tenantCfgCtx);
+                                if (carbonContext.getTenantId() != MultitenantConstants.INVALID_TENANT_ID) {
+                                    // Terminating idle tenant configuration contexts.
+                                    terminateTenantConfigContext(tenantCfgCtx);
+                                } else {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("If the tenant ID from the carbon context is -1, it is possible" +
```

Comment:
"Tenant ID of tenant domain " + tenantDomain + " can't be resolved. This tenant may be deleted. So skipping the tenant unloading part."